src/style/style_layer.js

```diff
@@ -155,13 +155,13 @@ class StyleLayer extends Evented {
             // so a call to _updateLayer is necessary, and we return true from this function so it gets called in
             // Style#setPaintProperty
             const prop = this._transitionablePaint._values[name];
-            const newCrossFadedValue = prop.property.specification["property-type"] === 'cross-faded-data-driven' && !prop.value.value && value;
+            const isCrossFadedValue = prop.property.specification["property-type"] === 'cross-faded-data-driven';
```

Comment:
Strictly speaking, the variable is representing `isCrossFadedProperty`.

Reply:
Ohhhh yes thank you. Was following the existing name but I agree with you.